### PR TITLE
Made Bound Controls Overridable And Updated Changelog

### PR DESCRIPTION
**Pull request by HossamBasion**
Made four functions in boundscontrol.cs protected virtual and updated changelog and package.json accordingly.

This is necessary for adding custom translation, rotation and scaling logic for bound controls behavior.

### CODE REVIEW

File: org.mixedrealitytoolkit.spatialmanipulation/CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
-## [3.2.0-development] - 2024-03-20
```
Comment (marlenaklein-msft) [inline org.mixedrealitytoolkit.spatialmanipulation/CHANGELOG.md:5]: please add a new section for 3.2.1 and add a bullet point to the added section to describe the change (can look to other package's change logs for example)

File: org.mixedrealitytoolkit.spatialmanipulation/BoundsControl/BoundsControl.cs [hunk 2]
```diff
@@ -683,7 +683,7 @@ private bool ComputeBounds(bool isSecondPass = false)
         /// Called by <see cref="BoundsHandleInteractable"/> from its OnSelectExited.
         /// Routes the XRI event data through, as well as a reference to itself, the selected handle.
         /// </summary>
-        internal void OnHandleSelectExited(BoundsHandleInteractable handle, SelectExitEventArgs args)
```
Comment (marlenaklein-msft) [inline org.mixedrealitytoolkit.spatialmanipulation/BoundsControl/BoundsControl.cs:686]: protected virtual internal? BoundsHandleInteractable needs to be able to access this

File: org.mixedrealitytoolkit.spatialmanipulation/BoundsControl/BoundsControl.cs [hunk 3]
```diff
@@ -698,7 +698,7 @@ internal void OnHandleSelectExited(BoundsHandleInteractable handle, SelectExitEv
         /// Called by <see cref="BoundsHandleInteractable"/> from its OnSelectEntered.
         /// Routes the XRI event data through, as well as a reference to itself, the selected handle.
         /// </summary>
-        internal void OnHandleSelectEntered(BoundsHandleInteractable handle, SelectEnterEventArgs args)
```
Comment (marlenaklein-msft) [inline org.mixedrealitytoolkit.spatialmanipulation/BoundsControl/BoundsControl.cs:701]: protected virtual internal?

Comment (marlenaklein-msft) [reply org.mixedrealitytoolkit.spatialmanipulation/BoundsControl/BoundsControl.cs:701]: does it compile with just protected virtual?

**Review by marlenaklein-msft — CHANGES_REQUESTED** [review]

**Review by marlenaklein-msft — DISMISSED** [review]

**Review by marlenaklein-msft — APPROVED** [review]

**Comment by AMollis** [comment]
@HossamBasion I need to review this change.  I'm not sure this is best solution for adding custom translation, rotation and scaling logic for bound controls behavior. 

I would like to see something similar to how we handle customization in the object manipulation behavior.


I'm going to review this and open bugs/issues as needed.

**Pull request merged**